### DISCUSSION TITLE
How to disable culture folders generated

### DISCUSSION THREAD

**Discussion by RaoHammas** [microsoft/microsoft-ui-xaml#8524] [Q&A]
Hi,
When I build the WinUI project, I see a lot of culture folders generated by this `Microsoft.ui.xaml.dll.mui`. 
I've tried different ways but none seems to work. I've tried these two properties but still, all folders are being generated.  

```
<SatelliteResourceLanguages>en-us</SatelliteResourceLanguages>
<NeutralLanguage>en-us</NeutralLanguage>
```  
How can I disable these folders generation? 
Thanks.

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/16499626/a9cafabf-7f21-4c5d-b673-a99b0ac62032)

**Comment by DarranRowe** [comment]
These aren't generated.
If you look in the Windows App SDK framework package (this is just a .zip file with a .msix extension), then you should see contents like the following:

![Screenshot 2023-06-06 200501](https://github.com/microsoft/microsoft-ui-xaml/assets/52577874/f05753d3-efd5-4c08-b6c3-7ae986026282)

Now, when you set your project to self contained, it needs to copy the files from the framework package to your application directory to satisfy dependencies. But this copying process is basically a dumb copy of all the non appx/msix files from the package.
The important thing about these files is that they use the same native resource format as Windows does.

![Screenshot 2023-06-06 201308](https://github.com/microsoft/microsoft-ui-xaml/assets/52577874/216de3a4-26d4-44a2-8d38-19cf1309bb91)

This means you can infer some behaviour as to how things work. Windows will use the Multilingual User Interface functions to access these files, and as you can see from Windows itself, only the relevant ones to your set languages are installed. So, if you are willing and daring and have no issues testing your application in all possible ways to make sure that things keep working, it may be possible to delete the languages that you don't need. But again, this should be technically taken as unsupported since the Windows App SDK groups these directories together.
This is a long way from not copying the files in the first place, but that would require changes to the build files, and I honestly don't think that would be something high on the list of things to do for the Windows App SDK.

**Reply by Balkoth** [reply]
I always thouth it is a really bad idea to it this way. You set attributes on which language you support and then have random non-supported languages in the output. I opened a ticket [here](https://developercommunity.visualstudio.com/t/It-is-not-possible-to-select-which-langu/10149123) which just got ignored, so you may be sol too.

**Comment by ghost1372** [comment]
I used this method:
First, create a file with the following name and extension:
**RemoveLanguageFolders.targets**
Then copy the following codes into this file:

```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">

  <!-- Define the OutputLanguages property -->
  <PropertyGroup>
    <OutputLanguages></OutputLanguages>
  </PropertyGroup>

  <!-- Split the ExcludedLanguages property into individual items -->
  <ItemGroup>
    <LanguageItems Include="$(ExcludedLanguages.Split(';'))" />
  </ItemGroup>

  <!-- Create the final output -->
  <Target Name="CreateOutput" AfterTargets="Build">
    <PropertyGroup>
      <OutputLanguagesToDelete>@(LanguageItems->'$(OutDir)%(Identity)\*.mui')</OutputLanguagesToDelete>
    </PropertyGroup>
  </Target>

  <!-- Remove Languages Folders -->
  <Target Name="RemoveFolders" AfterTargets="Build">
    <ItemGroup>
      <RemovingFiles Include="$(OutDir)*\*.mui" Exclude="$(OutputLanguagesToDelete)" />
      <RemovingFolders Include="@(RemovingFiles->'%(RootDir)%(Directory)')" />
    </ItemGroup>
    <RemoveDir Directories="@(RemovingFolders)" />
  </Target>

</Project>

```

now open your csproj file and import **RemoveLanguageFolders.targets** file:

`<Import Project="RemoveLanguageFolders.targets" />
`

you should edit **Project="RemoveLanguageFolders.targets"** if your file created in another place!

Now you can specify which folders should not be deleted, all folders will be automatically deleted unless specified with the following code:

```
  <PropertyGroup>
       ...........................
    <ExcludedLanguages>en-US;fa-IR</ExcludedLanguages>
  </PropertyGroup>
```
